### DISCUSSION TITLE
`yarn Install/dedupe` running on every command?

### DISCUSSION THREAD

**Discussion by xenobytezero** [moonrepo/moon#559] [Q&A] [answered]
Hey all,

Poking around in Moon as it looks like it's going to solve a bunch of problems I have with my project.

One of the big things I'm noticing is that every task I run is also running `yarn install` and `yarn dedupe`, even if the outputs have not changed.

Is this intended? Is there a way to skip this step for some commands? It's making my build times way longer for simple changes.

**Comment by milesj** [comment, marked as answer]
@xenobytezero Which version are you using? We just released v0.22 which should resolve this. If not, could you send over the `--log trace` output?

**Reply by castarco** [reply]
Is it possible that the same problem still remains for PNPM? I arrived here because of a similar issue, but with PNPM instead of Yarn.

**EDIT:** Nevermind, I found https://moonrepo.dev/api/types/interface/NodeConfig#dedupeOnLockfileChange

**Reply by milesj** [reply]
I believe it will also run if a package.json is currently modified in the git working tree, because its a touched/changed file.

**Comment by xenobytezero** [comment]
Yep, 0.22.0 looks like it solved it, thanks!

**Reply by milesj** [reply]
Awesome, glad to hear!
